### PR TITLE
Fix IntegrationTests verbosity issues

### LINKED ISSUE #78: IntegrationTests verbosity issues
The `IntegrationTests` test case has two logging issues:

- it lacks useful debug messages because `test_it` runs `simp_le` without the `--verbose` arg.
- each `self._run()` after the first one in `test_it`  will run through `setup_logging()` again and [add another handler to the root logger](https://github.com/zenhack/simp_le/blob/73853af828445f164f8b90f6f486b3d659012db8/simp_le.py#L1461), duplicating the messages.

That's why we end up with a log that look like this in Travis (there are currently five `self._run()`):

```
2017-10-05 02:24:53,344:INFO:simp_le:449: Saving key.pem
2017-10-05 02:24:53,344:INFO:simp_le:449: Saving key.pem
2017-10-05 02:24:53,344:INFO:simp_le:449: Saving key.pem
2017-10-05 02:24:53,344:INFO:simp_le:449: Saving key.pem
2017-10-05 02:24:53,344:INFO:simp_le:449: Saving key.pem
2017-10-05 02:24:53,345:INFO:simp_le:449: Saving full.pem
2017-10-05 02:24:53,345:INFO:simp_le:449: Saving full.pem
2017-10-05 02:24:53,345:INFO:simp_le:449: Saving full.pem
2017-10-05 02:24:53,345:INFO:simp_le:449: Saving full.pem
2017-10-05 02:24:53,345:INFO:simp_le:449: Saving full.pem
ok
----------------------------------------------------------------------
Ran 1 test in 4.748s
OK
```

I tried to add `-v/--verbose` to the test case without addressing the duplicate handler issue, as expected it's very unpleasant to parse.

Python 3 `logging` has a `hasHandlers()` method that allow to check for an existing handler before adding another, it works well but it's absent from Python 2. And I'm not sure that would be a proper way to address the issue in the first place.

### PR DESCRIPTION
**Pull request by buchdag**
This fix #78 by replacing calls to main() with subprocess.call() as discussed on the issue.

### CODE REVIEW

File: simp_le.py [hunk 1]
```diff
@@ -1672,15 +1673,15 @@ def test_it(self):
...
+            self.assertEqual(EXIT_REVOKE_OK, self._run([
+                "simp_le", "-v", "--server", (self.SERVER), "--revoke",
+                "-f", "account_key.json", "-f", "full.pem"]))
             # Revocation shouldn't touch any files
             self.assertEqual(initial_stats, self.get_stats(*files))
 
             # Changing SANs should trigger "renewal"
-            self.assertEqual(
-                EXIT_RENEWAL, self._run('%s -d le2.wtf:%s' % (args, webroot)))
+            args.extend(["-d", "le2.wtf:%s" % (webroot)])
```
Comment (zenhack) [inline simp_le.py]: Rather than using extend here, I'd rather just concatinate the lists, i.e.

```python
self.assertEqual(EXIT_RENEWAL, self._run(args + ["-d", "le2.wtf%s" % webroot]))
```

Rationale: mutating the list in-place makes the code more brittle, as the order of thee tests affects what arguments get passed. The same goes for the other mutations.

File: simp_le.py [hunk 2]
```diff
@@ -1649,19 +1650,19 @@ def _single_path_stats(path):
 
     def test_it(self):
         webroot = os.path.join(os.getcwd(), 'public_html')
-        args = ('--server %s --tos_sha256 %s -f account_key.json '
-                '-f key.pem -f full.pem -d le.wtf:%s' % (
-                    self.SERVER, self.TOS_SHA256, webroot))
+        args = ["simp_le", "-v", "--server", (self.SERVER), "--tos_sha256",
```
Comment (zenhack) [inline simp_le.py]: I think changing the name of the variable to `cmd` would make this a bit more clear, as now this includes the full command, not just the args. Same thing elsewhere.

**Comment by buchdag** [comment]
Tell me if the changes are good for you and I'll rebase them in the first commit.

**Comment by zenhack** [comment]
Works for me.

**Comment by buchdag** [comment]
Rebased.
